Mathlib/RingTheory/Grassmannian.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2025 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import Mathlib.RingTheory.Spectrum.Prime.FreeLocus
+
+/-!
+# Grassmannians
+
+## Main definitions
+
+- `Grassmannian`: `G(k, M; R)` is the `k`ᵗʰ Grassmannian of the `R`-module `M`. It is defined to be
+  the set of submodules of `M` whose quotient is locally free of rank `k`. Note that this indexing
+  is the opposite of some indexing in literature, where this rank would be `n - k` instead, where
+  `M = R ^ n`.
+
+## Implementation notes
+
+Convention for a finite dimensional vector space `V` over a field `F`, is that `G(k, V; F)` should
+parametrise `k`-dimensional subspaces of `V`, but this only works for finite dimensional spaces, and
```

Comment:
I think people will find this statement somewhere between dubious and triggering. I can certainly imagine some set of assumptions where I can make the Grassmannian as sub-objects beyond this. Less controversial would be saying Grothendieck follows this convention and so should we, perhaps hinting at why or not.